liquibase-standard/src/main/java/liquibase/Liquibase.java

```diff
@@ -1178,71 +1178,38 @@ public void reportStatus(boolean verbose, Contexts contexts, LabelExpression lab
         });
     }
 
+    @Deprecated
     public Collection<RanChangeSet> listUnexpectedChangeSets(String contexts) throws LiquibaseException {
         return listUnexpectedChangeSets(new Contexts(contexts), new LabelExpression());
     }
 
+    @Deprecated
     public Collection<RanChangeSet> listUnexpectedChangeSets(Contexts contexts, LabelExpression labelExpression)
             throws LiquibaseException {
-        changeLogParameters.setContexts(contexts);
-        changeLogParameters.setLabels(labelExpression);
-
-        ExpectedChangesVisitor visitor = new ExpectedChangesVisitor(database.getRanChangeSetList());
-
-        runInScope(() -> {
-
-            DatabaseChangeLog changeLog = getDatabaseChangeLog();
-            changeLog.validate(database, contexts, labelExpression);
-
-            ChangeLogIterator logIterator = new ChangeLogIterator(changeLog,
-                    new ContextChangeSetFilter(contexts),
-                    new LabelChangeSetFilter(labelExpression),
-                    new DbmsChangeSetFilter(database),
-                    new IgnoreChangeSetFilter());
-            logIterator.run(visitor, new RuntimeEnvironment(database, contexts, labelExpression));
-
-        });
-        return visitor.getUnexpectedChangeSets();
+        return UnexpectedChangesetsCommandStep.listUnexpectedChangeSets(getDatabase(), getDatabaseChangeLog(), contexts, labelExpression);
     }
 
-
+    @Deprecated
     public void reportUnexpectedChangeSets(boolean verbose, String contexts, Writer out) throws LiquibaseException {
         reportUnexpectedChangeSets(verbose, new Contexts(contexts), new LabelExpression(), out);
     }
 
+    @Deprecated
     public void reportUnexpectedChangeSets(boolean verbose, Contexts contexts, LabelExpression labelExpression,
                                            Writer out) throws LiquibaseException {
         changeLogParameters.setContexts(contexts);
         changeLogParameters.setLabels(labelExpression);
-        checkLiquibaseTables(false, getDatabaseChangeLog(true), null, null);
```

Comment:
Seems we lost checkLiquibaseTables when moving the code to the command. In fact a lot of steps use it, couldn't we create a helper for it?

Reply:
The command calls that method, here: https://github.com/liquibase/liquibase/blob/master/liquibase-standard/src/main/java/liquibase/command/core/helpers/DatabaseChangelogCommandStep.java#L85

Reply:
Thanks, I missed that.